conans/client/cmd/search.py

```diff
@@ -44,7 +44,7 @@ def search_packages(self, reference=None, remote=None, query=None, outdated=Fals
             remote = RemoteRegistry(self._client_cache.registry, self._user_io.out).remote(remote)
             packages_props = self._remote_manager.search_packages(remote, reference, query)
             ordered_packages = OrderedDict(sorted(packages_props.items()))
-            manifest = self._remote_manager.get_conan_digest(reference, remote)
+            _, manifest = self._remote_manager.get_conan_manifest(reference, remote)
```

Comment:
Because of the `b)`, the server always returns the resolved reference now (from remote manager)